lib/mosaic-routing/src/main/java/org/eclipse/mosaic/lib/routing/graphhopper/GraphHopperRouting.java

```diff
@@ -226,7 +265,8 @@ private EdgeFilter createEdgeFilterForRoutingPosition(final RoutingPosition posi
      */
     private Node determinePrefixNode(final RoutingPosition source) {
         if (source.getConnectionId() != null) {
-            return db.getConnection(source.getConnectionId()).getFrom();
+            List<Node> nodes = db.getConnection(source.getConnectionId()).getNodes();
+            return nodes.get(Math.max(0, nodes.size() - 2));
```

Comment:
What's the reasoning to use this node index?

Reply:
I'll add a comment. Actually we have more than 2 nodes per connection in the usual case.